### ISSUE TITLE
Valid semver string  1.0.2300254001 is failing validation

### ISSUE THREAD

**Issue by sohirani** [maxhauser/semver#79]
Semver 1.0.2300254001 is failing the validations. The patch version being generated with date and time stamp is a common scenario.

**Comment by WalkerCodeRanger** [comment]
This is being covered by issue #73 and will be fixed in v3.0.0

**Comment by sohirani** [comment]
Could you provide the estimated time when v3.0.0 will be available?

**Comment by damepatr** [comment]
@WalkerCodeRanger - Sorry, one last question. Do you plan on fixing this in #73? This issue got marked as a duplicate, but I can't tell if #73 includes patch numbers.

**Comment by WalkerCodeRanger** [comment]
@sohirani and @damepatr, I don't have an estimated time for when v3.0.0 will be available.  I've learned that for an open-source project like this, I just can't commit to when I'll have to time to finish a version. As a major release, it will take a while to develop. What I can say is that it is already under development, and the code currently on `master` is for v3.0.0. I've already addressed limits for prerelease identifiers on `master`.

@damepatr yes, I plan to address this as part of #73. While the title of that ticket only refers to prerelease identifiers, we discussed patch numbers there as well. They are fundamentally the same issue in different places. Namely, that `int` was used to represent all integer parts of the version number. Contributions are welcome on this issue. However, this is a major change. The planned change is to transition the `Major`, `Minor`, and `Patch` versions to `BigInteger`. So, I don't know if you want to try to tackle that.

I do plan to work on this next. Probably even a little tonight. As I said, it has already been addressed for prerelease identifiers. Since there is so much interest, I will plan to release a prerelease version of v3.0.0 with this fixed as soon as it is fixed.
